packages/storybook/src/generators/configuration/configuration.ts

```diff
@@ -48,16 +53,31 @@ export async function configurationGenerator(
   });
   tasks.push(initTask);
 
-  createRootStorybookDir(tree, schema.js, schema.tsConfiguration);
-  createProjectStorybookDir(
-    tree,
-    schema.name,
-    schema.uiFramework,
-    schema.js,
-    schema.tsConfiguration,
-    !!nextBuildTarget,
-    compiler === 'swc'
-  );
+  if (projectIsNested(root, sourceRoot, projectType)) {
```

Comment:
I don't think we can rely on `src` and `project.json`  being there, even though this is how we are generated our nested projects right now.

Maybe instead of checking for nested projects, we can compare the project root with workspace root?

Pseudo-code:
```ts
import { workspaceRoot } from 'nx/src/utils/workspace-root';

// ...
const projectDirectory = readProjectDirectory(options);
if (workspaceRoot === projectDirectory) {
  createRootStorybookDirForNested(...);
} else {
 // ...
}
  
}
```

Reply:
Aha! That looks much nicer! :)